pkgs/servers/monitoring/prometheus/cups-exporter.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, fetchFromGitHub, python3Packages }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "prometheus-cups-exporter-unstable";
+  version = "2019-03-17";
+
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "ThoreKr";
+    repo = "cups_exporter";
+    rev = "8fd1c2517e9878b7b7c73a450e5e546f437954a9";
+    sha256 = "1cwk2gbw2svqjlzgwv5wqzhq7fxwrwsrr0kkbnqn4mfb0kq6pa8m";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ prometheus_client pycups ];
+
+  installPhase = ''
+    mkdir -p $out/share/
+    cp cups_exporter.py $out/share/
+  '';
+
+  fixupPhase = ''
+    makeWrapper "${python3Packages.python.interpreter}" "$out/bin/prometheus-cups-exporter" \
+          --set PYTHONPATH "$PYTHONPATH" \
+          --add-flags "$out/share/cups_exporter.py"
+  '';
+
+  meta = with lib; {
+    description = "A simple prometheus exporter for cups implemented in python";
+    homepage = "https://github.com/ThoreKr/cups_exporter";
+    license = licenses.free;
```

Comment:
Unfortunately upstream doesn't specify any license, so it seems to me that setting the license to free, without having the copyright owner explicitly place the software in the public domain, wouldn't be correct.
Please correct me if I'm wrong, I'm not a lawyer :)

Otherwise this looks good to me :+1:

Reply:
Oh damn, I forgot to change it to unfree. I actually opened an issue with upstream asking for one. So until it is done, unfree be it :)